src/rofl/common/cthread.cpp

```diff
@@ -202,9 +363,44 @@ void cthread::drop_read_fd(int fd, bool exception) {
     };
     }
   }
+
+  if (events_reg[fd]) {
+    AcquireReadWriteLock lock(events_lock);
+    io_event *ioev = static_cast<io_event *>(events_reg[fd]);
+
+    if (ioev->we) {
+      ioev->re = nullptr;
+    } else {
+      delete events_reg[fd];
+      events_reg[fd] = nullptr;
+    }
+  }
 }
 
-void cthread::add_write_fd(int fd, bool exception, bool edge_triggered) {
+void cthread::add_write_fd(int fd, cthread_write_event *cb, void *userdata,
+                           bool exception, bool edge_triggered) {
+
```

Comment:
For the registration it should not matter since the event loop cannot deal with the structure yet. I will double check.